site/src/pages/TerminalPage/TerminalPage.tsx

```diff
@@ -127,13 +129,13 @@ const TerminalPage: FC = () => {
     window.addEventListener("resize", listener);
 
     // Terminal is correctly sized and is ready to be used.
-    terminalRef.current = terminal;
+    setTerminal(terminal);
 
     return () => {
       window.removeEventListener("resize", listener);
       terminal.dispose();
     };
-  }, [theme, renderer, config.isLoading, terminalWrapperRef, handleWebLinkRef]);
+  }, [config.isLoading, renderer, theme.palette.background.default]);
```

Comment:
I think I just realized something: I don't think we need the `config.isLoading` value in the dependency array. The main thing we care about is whether we have a `renderer` value from the config query, and that's already part of the array

So I think we can get away with removing `config.isLoading` and updating the guard clause at the top of the effect to:
```tsx
if (!terminalWrapperRef.current || !renderer) {
  return;
}
```
I don't know about how often the config value changes, or what can change it, but I also feel like doing it like this could also insulate us better against RQ background refetches

Reply:
Makes sense, let me take a look.